File: app/src/main/java/ru/merkulyevsasha/news/NewsApp.java
package ru.merkulyevsasha.news;

import android.app.Activity;
import android.app.Application;
import android.app.Service;

import com.evernote.android.job.JobManager;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;
import dagger.android.HasServiceInjector;
import ru.merkulyevsasha.news.dagger.components.AppComponent;
import ru.merkulyevsasha.news.dagger.components.DaggerAppComponent;

/**
 * Created by sasha_merkulev on 21.09.2017.
 */

public class NewsApp extends Application
        implements HasActivityInjector, HasServiceInjector {

    @Inject DispatchingAndroidInjector<Activity> activityInjector;
    @Inject DispatchingAndroidInjector<Service> serviceInjector;

    @Override
    public void onCreate() {
        super.onCreate();

        AppComponent component = DaggerAppComponent
                .builder()
                .context(this)
                .build();

        component.inject(this);

        JobManager.create(this).addJobCreator(component.getJobCreator());
    }

    @Override
    public AndroidInjector<Activity> activityInjector() {
        return activityInjector;
    }

    @Override
    public AndroidInjector<Service> serviceInjector() { return serviceInjector; }

}

File: app/src/main/java/ru/merkulyevsasha/news/NewsApp.kt
package ru.merkulyevsasha.news

import android.app.Application
import android.os.StrictMode
import android.os.StrictMode.VmPolicy
import com.facebook.stetho.Stetho
import com.google.android.gms.ads.MobileAds
import com.squareup.leakcanary.LeakCanary
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class NewsApp : Application() {

    override fun onCreate() {
        super.onCreate()
        if (LeakCanary.isInAnalyzerProcess(this)) {
            // This process is dedicated to LeakCanary for heap analysis.
            // You should not init your app in this process.
            return;
        }
        LeakCanary.install(this);


        MobileAds.initialize(this, getString(R.string.APP_ID))

        if (BuildConfig.DEBUG_MODE) {
            Stetho.initializeWithDefaults(this)

//            StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder()
//                .detectDiskReads()
//                .detectDiskWrites()
//                .detectNetwork()   // or .detectAll() for all detectable problems
//                .penaltyLog()
//                .build())
//
//            StrictMode.setVmPolicy(VmPolicy.Builder()
//                .detectLeakedSqlLiteObjects()
//                //.detectLeakedClosableObjects()
//                .penaltyLog()
//                .penaltyDeath()
//                .build())
        }

    }
}
